### PR TITLE
FAT-1009 Loans: Item with Restricted status should change status to Available after it is checked in

### PR DESCRIPTION
**Pull request by bisecomsergiy**
### Purpose
Increase mod-circulation coverage by integration tests
### Approach
Created karate test for Loans: When an item that had the status of restricted that was checked out is checked in, set the item status to Available
### Learning
https://issues.folio.org/browse/FAT-1009

### CODE REVIEW

File: mod-circulation/src/main/resources/domain/mod-circulation/circulation-junit.feature [hunk 1]
```diff
@@ -62,6 +62,7 @@ Feature: mod-circulation integration tests
       | 'owners.item.post'                                             |
       | 'users.item.post'                                              |
       | 'usergroups.item.post'                                         |
+      | 'inventory.items.item.mark-restricted.post'                    |
```
Comment (viktor-draban) [inline mod-circulation/src/main/resources/domain/mod-circulation/circulation-junit.feature:65]: Please sort a permissions based on alphabet

File: mod-circulation/src/main/resources/domain/mod-circulation/features/loans.feature [hunk 2]
```diff
@@ -377,3 +377,38 @@ Feature: Loans tests
     And match error.message == '#string'
     And match error.message == 'Long Way to a Small Angry Planet (' + materialTypeName + ') (Barcode: ' + extItemBarcode + ') has the item status Intellectual item and cannot be checked out'
     And match error.parameters[0].value == extItemBarcode
+
+  Scenario: When an item that had the status of restricted that was checked out is checked in, set the item status to Available
+
+    * def itemBarcode = '88888'
```
Comment (viktor-draban) [inline mod-circulation/src/main/resources/domain/mod-circulation/features/loans.feature:383]: add userBarcode explicitly

File: mod-circulation/src/main/resources/domain/mod-circulation/features/loans.feature [hunk 3]
```diff
@@ -377,3 +377,38 @@ Feature: Loans tests
...
+
+    # post associated entities and item
+    * call read('classpath:domain/mod-circulation/features/util/initData.feature@PostInstance')
+    * def postServicePointResult = call read('classpath:domain/mod-circulation/features/util/initData.feature@PostServicePoint')
+    * def servicePointId = postServicePointResult.response.id
```
Comment (viktor-draban) [inline mod-circulation/src/main/resources/domain/mod-circulation/features/loans.feature:388]:
```suggestion
    * call read('classpath:domain/mod-circulation/features/util/initData.feature@PostServicePoint')
```

File: mod-circulation/src/main/resources/domain/mod-circulation/circulation-junit.feature (hunk 1 above)
Comment (bisecomsergiy) [reply mod-circulation/src/main/resources/domain/mod-circulation/circulation-junit.feature]: Sure, it is done. Thanks

**Review by viktor-draban — APPROVED** [review]

File: mod-circulation/src/main/resources/domain/mod-circulation/features/loans.feature (hunk 3 above)
Comment (bisecomsergiy) [reply mod-circulation/src/main/resources/domain/mod-circulation/features/loans.feature]: Thank you, it is done..

**Pull request merged**